provider/internal/java/dependency.go

```diff
@@ -517,3 +539,75 @@ func loadDepLabelItems(r io.Reader, depToLabels map[string]*depLabelItem, label
 	}
 	return nil
 }
+
+// GetLocation given a dep, attempts to find line number, caches the line number for a given dep
+func (j *javaServiceClient) GetLocation(ctx context.Context, dep konveyor.Dep) (engine.Location, error) {
+	location := engine.Location{StartPosition: engine.Position{}, EndPosition: engine.Position{}}
+	// we know that this provider populates extras with required information
+	if dep.Extras == nil {
+		return location, fmt.Errorf("unable to get location for dep %s", dep.Name)
+	}
+	extrasKeys := []string{artifactIdKey, groupIdKey, pomPathKey}
+	for _, key := range extrasKeys {
+		if val, ok := dep.Extras[key]; !ok {
+			return location,
+				fmt.Errorf("unable to get location for dep %s, missing extras %s", dep.Name, key)
+		} else if _, ok := val.(string); !ok {
+			return location,
+				fmt.Errorf("unable to get location for dep %s, extras %s not a string", dep.Name, key)
+		}
+	}
+
+	cacheKey := fmt.Sprintf("%s-%s-%s-%v",
+		dep.Name, dep.Version, dep.ResolvedIdentifier, dep.Indirect)
+	if val, exists := j.depsLocationCache[cacheKey]; exists {
+		if val == 0 {
+			return location,
+				fmt.Errorf("unable to get location for dep %s due to a previous error", dep.Name)
+		}
+		return engine.Location{
+			StartPosition: engine.Position{
+				Line: val,
+			},
+			EndPosition: engine.Position{
+				Line: val,
+			},
+		}, nil
+	}
+	defer func() {
+		j.depsLocationCache[cacheKey] = location.StartPosition.Line
+	}()
+
+	groupId := dep.Extras[groupIdKey]
+	artifactId := dep.Extras[artifactIdKey]
+	path := dep.Extras[pomPathKey]
+	cmd := exec.CommandContext(ctx, "pcre2grep", []string{
```

Comment:
Why can't we use grep -E?

Reply:
@shawn-hurley main problem with grep is that it doesn't do a multiline search. Or at least, I couldn't find any litarature around that.

Reply:
heard, makes sense, just wondering

Reply:
@shawn-hurley had to remove usage of `pcregrep` as it is unavailable in ubi. wrote a small parser myself that respects the timeout context